packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -0,0 +1,301 @@
+import Foundation
+import EmbraceIO
+import EmbraceCore
+import EmbraceCrash
+import EmbraceCommonInternal
+
+#if canImport(CodePush)
+import CodePush
+#endif
+
+enum EmbraceKeys: String {
+  case reactNativeVersion = "io.embrace.reactnative.version"
+  case embraceReactNativeSdkVersion = "io.embrace.reactnative.sdk.version"
+  case javaScriptPatchNumber = "io.embrace.javascript.patch"
+  case javaScriptBundleURL = "io.embrace.jsbundle.url"
+}
+
+@objc(EmbraceManager)
+class EmbraceManager: NSObject {
+  @objc
+  func setJavaScriptBundlePath(_ path: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.javaScriptBundleURL.rawValue, value: path)
+      resolve(true)
+    }  catch let error {
+      reject("SET_JS_BUNDLE_PATH_ERROR", "Error setting JavaScript bundle path", error)
+    }
+  }
+  
+  @objc
+  func isStarted(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    if let embraceStarted = Embrace.client?.started {
+      resolve(embraceStarted)
+    } else {
+      resolve(false)
+    }
+  }
+  
+  @objc(startNativeEmbraceSDK:resolver:rejecter:)
```

Comment:
some of the `@objc` decorators have the method name + args but others are just the decorator, in what cases are they are required? Should we just always have them?

Reply:
When you have a parameter else than the promises, you need to specify it

Reply:
that seems inconsistent with the current methods though, e.g. `setUserIdentifier` below has the `userIdentifier` parameter but it only has the `@objc` decorator by itself?